android/src/main/java/com/google/recaptchaenterprisereactnative/RecaptchaEnterpriseReactNativeModule.kt

```diff
@@ -47,64 +48,76 @@ class RecaptchaEnterpriseReactNativeModule(reactContext: ReactApplicationContext
       return RecaptchaAction.custom(actionStr)
     }
   }
-
+  
   @ReactMethod
   fun fetchClient(siteKey: String, promise: Promise) {
     GlobalScope.launch {
       try {
-          recaptchaClient = Recaptcha.fetchClient(application, siteKey)
-          promise.resolve(null)
-      }catch (exception: Exception){
-          promise.reject(exception)
+        recaptchaClient = Recaptcha.fetchClient(application, siteKey)
+        promise.resolve(null)
+      } catch (exception: Exception) {
+        promise.reject(exception)
```

Comment:
Should we do here the same that we are doing in the `.onFailure` of the `initClient` and use `promise.reject` passing an specific errorCode?

Reply:
Agreed, missed that one.  I went ahead and added tests for fetch/get client errors using a bad sitekey. It is a littl bit wrong because there is no reset in the SDK and the detox tests don't fully reset the app.  So the tests are order dependent unfortunately until we add multi-tenancy and/or reset.